ansible/roles/nginx/vars/production.yml

```diff
@@ -0,0 +1,96 @@
+---
+ubuntu_pkg: nginx
+
+nginx_sites:
+- server:
+   file_name: "{{ deploy_env }}_commcare_public"
+   listen: "80"
+   server_name: "{{ WWW_SITE_HOST }}"
+   client_max_body_size: 100m
+   use_balancer: True
+   proxy_set_header: "Host $http_host"
+   location1:
+    name: /
+    return: "301 https://$server_name$request_uri"
+   location2:
+    name: "~* /(?:home|users|support|plans|images|pdfs|style|js|tour|div2|docs|fr|poc|es)/"
+    alias: "{{ public_site_path }}/$1"
+    try_files: "$uri $uri/index.html $uri/ =404"
+- server:
+   file_name: "{{ deploy_env }}_commcare_public_redirect"
+   listen: "80"
+   server_name: "{{ WWW_SITE_HOST }}"
+   client_max_body_size: 100m
+   proxy_set_header: "Host $http_host"
+   location1:
+    name: /
+    return: "301 http://{{ SITE_HOST }}$request_uri"
+- server:
+   file_name: "{{ deploy_env }}_commcare"
+   listen: "443 ssl"
+   server_name: "{{ WWW_SITE_HOST }}"
+   client_max_body_size: 100m
+   use_balancer: True
+   proxy_set_header: "Host $http_host"
+   location1:
+    name: /
+    # The /dev/null is a hack because try files require more than one argument,
+    # might make sense to remove the @proxy_to_app and just use proxy_pass
+    try_files: "/dev/null @proxy_to_app"
+   location2:
+    name: "@proxy_to_app"
+    use_balancer: True
+   location3:
+    name: "~* /(?:home|users|support|plans|images|pdfs|style|js|tour|div2|docs|fr|poc|es)/"
+    alias: "{{ public_site_path }}/$1"
+    try_files: "$uri $uri/index.html $uri/ =404"
+   # I am having a hard time telling how the locations
+   location4:
+    name: "~* /static"
+    alias: "{{ nginx_static_home }}"
+    try_files: "$uri $uri/index.html $uri/ =404"
+- server:
+   file_name: "{{ deploy_env }}_commcare_redirect"
+   listen: "443 ssl"
+   server_name: "{{ SITE_HOST }}"
+   client_max_body_size: 100m
+   proxy_set_header: "Host $http_host"
+   location1:
+    name: /
+    return: "301 https://{{ WWW_SITE_HOST }}$request_uri"
+- server:
+   file_name: commtrack_public
+   listen: "8080"
+   server_name: "{{ COMMTRACK_SITE_HOST }}"
+   client_max_body_size: 100m
+   use_balancer: True
+   proxy_set_header: "Host $http_host"
+   location1:
+    name: /
+    return: "301 https://$server_name$request_uri"
+   location2:
+    name: "~* /(?:home|users|support|plans|images|pdfs|style|js|tour|div2|docs|fr|poc|es)/"
+    alias: "{{ public_site_path }}/$1"
+    try_files: "$uri $uri/index.html $uri/ =404"
+- server:
+   file_name: commtrack
+   listen: "8443 ssl"
```

Comment:
Yeah, looks like you saw this, but Commtrack hits a separate proxy before it goes to hqproxy.